pennylane/ops/qubit/qchem_ops.py

```diff
@@ -1055,3 +1055,192 @@ def compute_decomposition(phi, wires):
+        &|01\rangle \mapsto e^{i \phi/2} \cos(\phi/2)|01\rangle - ie^{i \phi/2} \sin(\phi/2)|10\rangle\\
+        &|10\rangle \mapsto -ie^{i \phi/2} \sin(\phi/2)|01\rangle + e^{i \phi/2} \cos(\phi/2)|10\rangle\\
+        &|11\rangle \mapsto e^{i \phi}|11\rangle,
+
+    where a qubit in :math:`|0\rangle` and :math:`|1\rangle` states represent a hole and a fermion in
+    the orbital, respectively. It preserves anti-symmetrization of orbitals by applying a phase factor
+    of :math:`e^{i \phi/2}` to the state for each qubit initially in :math:`|1\rangle` state. Consequently,
+    for :math:`\phi=\pi`, the given rotation will essentially perform a SWAP operation on the qubits while
+    applying a global phase of :math:`-1`, if both qubits are :math:`|1\rangle`.
+
+    **Details:**
+
+    * Number of wires: 2
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+    * Gradient recipe: :math:`\frac{d}{d\phi}f(U(\phi)) = \frac{1}{2}\left[f(U(\phi+\pi/2)) - f(U(\phi-\pi/2))\right]`
+      where :math:`f` is an expectation value depending on :math:`U(\phi)`
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int]): the wires the operation acts on
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+
+    **Example**
+
+    The following circuit performs the transformation: :math:`|01\rangle \mapsto e^{i \phi/2}
+    \cos(\phi/2)|01\rangle - ie^{i \phi/2} \sin(\phi/2)|10\rangle`, where :math:`\phi=0.1`:
+
+    .. code-block::
+
+        >>> dev = qml.device('default.qubit', wires=2)
+        >>> @qml.qnode(dev)
+        ... def circuit(phi):
+        ...     qml.PauliX(wires=1)
+        ...     qml.FermionicSWAP(phi, wires=[0, 1])
+        ...     return qml.state()
+        >>> circuit(0.1)
+        array([0.+0.j, 0.9975+0.04992j, 0.0025-0.04992j, 0.+0.j])
+    """
+
+    num_wires = 2
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (0,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = "A"
+    """Gradient computation method."""
+
+    parameter_frequencies = [(1,)]
+    """Frequencies of the operation parameter with respect to an expectation value."""
+
+    def generator(self):
+        w1, w2 = self.wires
+        return 0.5 * qml.Identity(w1) @ qml.Identity(w2) - 0.25 * (
+            qml.Identity(w1) @ qml.PauliZ(w2)
+            + qml.PauliZ(w1) @ qml.Identity(w2)
+            + qml.PauliX(w1) @ qml.PauliX(w2)
+            + qml.PauliY(w1) @ qml.PauliY(w2)
+        )
+
+    def __init__(self, phi, wires, do_queue=True, id=None):
+        super().__init__(phi, wires=wires, do_queue=do_queue, id=id)
```

Comment:
this is fine but this is also handled inherently because you set `num_wires` and `num_params`, so it could be left out. this might result in a more informative error so I suppose it's alright too